applicationset/generators/git.go

```diff
@@ -222,17 +212,108 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 			}
 		}
 
-		err := appendTemplatedValues(values, params, useGoTemplate, goTemplateOptions)
+		err := appendTemplatedValues(git.Values, params, useGoTemplate, goTemplateOptions)
 		if err != nil {
 			return nil, fmt.Errorf("failed to append templated values: %w", err)
 		}
 
+		// Get the common parameters from files, removing duplicates
+		// We need to resolve them for each git file, as they could depend on parameters inside
+		err = g.appendExtraParams(git, goTemplateOptions, useGoTemplate, noRevisionCache, verifyCommit, params)
+		if err != nil {
+			return nil, err
+		}
+
 		res = append(res, params)
 	}
 
 	return res, nil
 }
 
+func (g *GitGenerator) appendExtraParams(git *argoprojiov1alpha1.GitGenerator, goTemplateOptions []string, useGoTemplate bool, noRevisionCache bool, verifyCommit bool, params map[string]interface{}) error {
+	extraParams, err := g.getExtraParams(git, goTemplateOptions, useGoTemplate, noRevisionCache, verifyCommit, params)
+	if err != nil {
+		return fmt.Errorf("failed to append extra parameters from files: %w", err)
+	}
+	if len(extraParams) == 0 {
+		return nil
+	}
+	if useGoTemplate {
```

Comment:
What's the meaning of `useGoTemplate`? Where is it set?

Reply:
It's an applicationset general option... Nothing new